libs/data/bbhnet/data/utils.py

```diff
@@ -0,0 +1,94 @@
+from typing import Optional
+
+import numpy as np
+
+
+def sample_kernels(
+    x: np.ndarray, size: int, N: Optional[int] = None
+) -> np.ndarray:
+    """Sample fixed-size kernels from a timeseries array
+
+    Given an array of timeseries `x`, whose 0th dimension
+    indexes independent samples and whos last dimension
+    indexes time, sample `N` kernels from among the samples
+    in `x` with length `size`. If `x` has only a single dimension,
+    `N` must be provided and `N` kernels will be sampled from
+    that single timeseries
+
+    Args:
+        x: array of timeseries to sample from
+        size: the size of kernels to sample from `x` (in samples)
+        N: The number of samples to generate. If left as `None`,
+            a kernel will be generated from each row in `x` in order
+            if `x.ndim > 1`, otherwise a `ValueError` will be
+            raised.
+    Returns:
+        Array of sampled kernels of size `(n, ..., size)`, where
+            `n = len(x) if N is None else N` and `...` represents
+            all of `x`'s intermediate dimensions between its first
+            and last.
+    """
+
+    if len(x.shape) == 1 and N is None:
+        raise ValueError(
+            "Must specify number of samples N when x is 1 dimensional"
+        )
+    elif len(x.shape) == 1:
+        # this is a 1D array, so we'll just be sampling
+        # the N kernels from the existing timeseries
+        if N > (len(x) - size):
+            raise ValueError(
+                "Array of length {} too short to sample {} "
+                "kernels of length {}".format(len(x), size, N)

```

Comment:
Should the N and size be switched in ValueError message: `.format(len(x), N, size)`